charts/cloudprem/README.md.gotmpl

```diff
@@ -0,0 +1,197 @@
+
+# Metastore configuration
+# The metastore is responsible for storing and managing index metadata.
+# It requires a PostgreSQL database connection string to be provided via a Kubernetes secret.
+# The secret should contain a key named `QW_METASTORE_URI` with a value in the format:
+# postgresql://<username>:<password>@<host>:<port>/<database>
+#
+# The metastore connection string is mounted into the pods using extraEnvFrom to reference the secret.
+metastore:
+  extraEnvFrom:
+    - secretRef:
+        name: cloudprem-metastore-uri
+
+# Indexer configuration
+# The indexer is responsible for processing and indexing incoming data it receives data from various sources (e.g., Datadog agents, log collectors)
+# and transforms it into searchable files called "splits" stored in S3.
+#
+# The indexer is horizontally scalable - you can increase `replicaCount` to handle higher indexing throughput.
+# Resource requests and limits should be tuned based on your indexing workload.
+#
+# The default values are suitable for moderate indexing loads of up to 20MB/s per indexer pod.
+indexer:
+  replicaCount: 2
+
+  resources:
+    requests:
+      cpu: "4"
+      memory: "8Gi"
+    limits:
+      cpu: "4"
+      memory: "8Gi"
+
+# Searcher configuration
+# The searcher is responsible for executing search queries against the indexed data stored in S3.
+# It handles search requests from Datadog's query service and returns matching results.
+#
+# The searcher is horizontally scalable - you can increase `replicaCount` to handle more concurrent searches.
+# Resource requirements for searchers are highly workload-dependent and should be determined empirically.
+# Key factors that impact searcher performance include:
+# - Query complexity (e.g., number of terms, use of wildcards or regex)
+# - Query concurrency (number of simultaneous searches)
+# - Amount of data scanned per query
+# - Data access patterns (cache hit rates)
+#
+# Memory is particularly important for searchers as they cache frequently accessed index data in memory.
+# Monitor searcher metrics and adjust resources based on observed performance and workload characteristics.
+searcher:
+  replicaCount: 2
+
+  resources:
+    requests:
+      cpu: "4"
+      memory: "16Gi"
+    limits:
+      cpu: "4"
+      memory: "16Gi"
+```
+
+### Installing or upgrading the Helm chart
+
+```sh
+helm upgrade --install <release name> datadog/cloudprem \
+  -n <namespace name> \
+  -f datadog-values.yaml
+```
+
+### Uninstalling the Helm chart
+To uninstall the deployment:
+
+```sh
+helm uninstall <release name>
+```
+
+This command removes all the Kubernetes resources associated with the chart and deletes the release.
+
+## Helm Chart values (non-exhaustive)
+
+| Key | Type | Default | Description
+| :--------------- |:---------------:| -----:|--- |
+|aws.accountId | string | null | AWS account ID used for the EKS role ARN service account annotation|
```

Comment:
nits:
In other charts, this table is generated from comments in the values.yaml file.
No need to do the same here, since it would require a significant number of comment updates in values.yaml.
That said, it could be a nice improvement in the future to ensure all options are documented and kept in sync with values.yaml.